adapter/internal/discovery/xds/rate_limiter_cache.go

```diff
@@ -257,9 +291,46 @@ func (r *rateLimitPolicyCache) updateXdsCache(label string) bool {
 	return true
 }
 
+// AddSubscriptionLevelRateLimitPolicy adds a subscription level rate limit policy to the cache. This method is called
+// only during the startup as there is no option available to add subscription level rate limit policies via the choreo console
+func AddSubscriptionLevelRateLimitPolicy(policyList *types.SubscriptionPolicyList) error {
+	// Check if rlsPolicyCache.metadataBasedPolicies[Subscription] exists and create a new map if not
+	if _, ok := rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType]; !ok {
+		rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType] = make(map[string]*rls_config.RateLimitDescriptor)
+	}
+	for _, policy := range policyList.List {
+		// Needs to skip on async policies.
+		if policy.DefaultLimit == nil || policy.DefaultLimit.QuotaType != "requestCount" || policy.DefaultLimit.RequestCount == nil {
```

Comment:
Is it possible to rename this variable to imply subscriptions ?

Reply:
I wonder if it is really necessary because method name already says addSubscriptionLevelPolicies. So there won't be any other type of policies here. wdyt?